src/go/rpk/pkg/config/schema.go

```diff
@@ -31,8 +31,8 @@ type RedpandaConfig struct {
 	KafkaApi                             []NamedSocketAddress   `yaml:"kafka_api" mapstructure:"kafka_api" json:"kafkaApi"`
 	AdvertisedKafkaApi                   []NamedSocketAddress   `yaml:"advertised_kafka_api,omitempty" mapstructure:"advertised_kafka_api,omitempty" json:"advertisedKafkaApi,omitempty"`
 	KafkaApiTLS                          []ServerTLS            `yaml:"kafka_api_tls,omitempty" mapstructure:"kafka_api_tls,omitempty" json:"kafkaApiTls"`
-	AdminApi                             SocketAddress          `yaml:"admin" mapstructure:"admin" json:"admin"`
-	AdminApiTLS                          ServerTLS              `yaml:"admin_api_tls,omitempty" mapstructure:"admin_api_tls,omitempty" json:"adminApiTls"`
+	AdminApi                             []NamedSocketAddress   `yaml:"admin" mapstructure:"admin" json:"admin"`
```

Comment:
i think this can be wrapped in a type. 

that gives first class methods for Port(), Connect(), or smth like that.

Reply:
Generally, I think that would make sense. For example,

```
type AdminAPI []NamedSocketAddress

func (*AdminAPI) getDefaultAddress() NamedSocketAddress {
   return findAddressByName("default")
}
etc.
```

While preparing this PR though I didn't see rpk calling the Admin API (the occurrence in your previous comment seems to be the only one and it's to retrieve a default value). Let me know if I'm missing something.

On the other hand, perhaps the Kafka API would benefit from such change. Should we consider this "typed" change separately - across APIs?

Also, @0x5d and @BenPope, what do you think?

Reply:
+1 for going to typed approach and in another PR. Could you open an issue so we can track it?

Reply:
#1137